app/views/posts/partials/show/_image.html.erb

```diff
@@ -1,3 +1,7 @@
 <% if policy(post).visible? %>
-  <%= image_tag(post.file_url_for(CurrentUser.user), width: post.image_width_for(CurrentUser.user), height: post.image_height_for(CurrentUser.user), id: "image", class: "fit-width", "data-original-width": post.image_width, "data-original-height": post.image_height, "data-large-width": post.large_image_width, "data-large-height": post.large_image_height, "data-tags": post.tag_string, alt: post.presenter.humanized_essential_tag_string, "data-uploader": post.uploader.name, "data-rating": post.rating, "data-flags": post.status_flags, "data-parent-id": post.parent_id, "data-has-children": post.has_children?, "data-has-active-children": post.has_active_children?, "data-score": post.score, "data-fav-count": post.fav_count, "itemprop": "contentUrl") %>
+  <%= content_tag(:picture) do -%>
+    <%= tag.source media: "(min-width: 660px)", srcset: post.tagged_file_url -%>
+    <%= tag.source media: "(max-width: 660px)", srcset: post.tagged_large_file_url -%>
+    <%= tag.img width: post.image_width_for(CurrentUser.user), height: post.image_height_for(CurrentUser.user), id: "image", class: "fit-width", "data-original-width": post.image_width, "data-original-height": post.image_height, "data-large-width": post.large_image_width, "data-large-height": post.large_image_height, "data-tags": post.tag_string, alt: post.presenter.humanized_essential_tag_string, "data-uploader": post.uploader.name, "data-rating": post.rating, "data-flags": post.status_flags, "data-parent-id": post.parent_id, "data-has-children": post.has_children?, "data-has-active-children": post.has_active_children?, "data-score": post.score, "data-fav-count": post.fav_count, itemprop: "contentUrl", src: post.file_url_for(CurrentUser.user) %>
```

Comment:
wouldn't it be more appropriate to put most of those attributes (including `id="img"`) on the `<picture>` tag? it would probably break some custom css but that's probably the case even as it is now.

Reply:
All of the image and data attributes added are all on the `#image` element, and having the ID also on that element makes it slightly easier for the users to access. Also, anything that gets done to the `<img>` element gets done to the image in general, so if the user does something like:

```javascript
$("#image").hide();
```

then the image will be hidden, even if if the original `src` set on the image isn't what is currently being shown. In fact, if there isn't an `<img>` element at all in that grouping, then the whole thing doesn't work at all. I learned that to my own detriment after trying it out myself. In fact, as I understand it, the image element is what always has the image loaded onto it, and the `<source>` elements are only changing the current `src` that is set on the `<img>` element.

Reply:
Ok it seems you're right about that, `<picture>` is a bit weird.
[The WhatWG says this in the HTML spec](https://html.spec.whatwg.org/multipage/embedded-content.html#the-picture-element):

> Also, the picture element itself does not display anything; it merely provides a context for its contained img element that enables it to choose from multiple URLs.

Apparently I made some of my custom css for image previews a lot more complicated than necessary.